app/models/trends/statuses.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+class Trends::Statuses < Trends::Base
+  PREFIX = 'trending_statuses'
+
+  self.default_options = {
+    threshold: 5,
+    review_threshold: 3,
+    score_halflife: 2.hours.freeze,
+  }
+
+  class Query < Trends::Query
+    def filtered_for!(account)
+      @account = account
+      self
+    end
+
+    def filtered_for(account)
+      clone.filtered_for!(account)
+    end
+
+    private
+
+    def apply_scopes(scope)
+      scope.includes(:account)
+    end
+
+    def perform_queries
+      return super if @account.nil?
+
+      statuses        = super
+      account_ids     = statuses.map(&:account_id)
+      account_domains = statuses.map(&:account_domain)
+
+      preloaded_relations = {
+        blocking: Account.blocking_map(account_ids, @account.id),
+        blocked_by: Account.blocked_by_map(account_ids, @account.id),
+        muting: Account.muting_map(account_ids, @account.id),
+        following: Account.following_map(account_ids, @account.id),
+        domain_blocking_by_domain: Account.domain_blocking_map_by_domain(account_domains, @account.id),
+      }
+
+      statuses.reject { |status| StatusFilter.new(status, @account, preloaded_relations).filtered? }
+    end
+  end
+
+  def register(status, at_time = Time.now.utc)
+    add(status.proper, status.account_id, at_time) if eligible?(status)
+  end
+
+  def add(status, _account_id, at_time = Time.now.utc)
+    # We rely on the total reblogs and favourites count, so we
+    # don't record which account did the what and when here
+
+    record_used_id(status.id, at_time)
+  end
+
+  def query
+    Query.new(key_prefix, klass)
+  end
+
+  def refresh(at_time = Time.now.utc)
+    statuses = Status.where(id: (recently_used_ids(at_time) + currently_trending_ids(false, -1)).uniq).includes(:account, :media_attachments)
+    calculate_scores(statuses, at_time)
+    trim_older_items
+  end
+
+  def request_review
+    statuses = Status.where(id: currently_trending_ids(false, -1)).includes(:account)
+
+    statuses.filter_map do |status|
+      next unless would_be_trending?(status.id) && !status.trendable? && status.requires_review_notification?
+
+      status.account.touch(:requested_review_at)
+      status
+    end
+  end
+
+  protected
+
+  def key_prefix
+    PREFIX
+  end
+
+  def klass
+    Status
+  end
+
+  private
+
+  def eligible?(status)
+    original_status = status.proper
+
+    original_status.public_visibility? &&
+      original_status.account.discoverable? && !original_status.account.silenced? &&
+      original_status.spoiler_text.blank? && !original_status.sensitive? && !original_status.reply?
```

Comment:
unlike trending links, where the CW is lost and thus leading to lost context, having CWed toots trending is probably fine; in fact, filtering out CWed toots for trendings my be an intensive to *not* use the feature

Reply:
We can't easily differentiate between CWs that are there because it's food and the author is conscious of people who don't want to see food, and CWs that are there because it's adult content. We err on the safe side here. Besides, browsing things you can't see and need to tap on is not the most satisfying experience.

Reply:
> We can't easily differentiate between CWs that are there because it's food and the author is conscious of people who don't want to see food, and CWs that are there because it's adult content. 

This is a bit tangential, but would you be open to a feature request to split the "CW" feature from the "mark as sensitive" feature? I know they're at least two separate fields in ActivityPub and also in the API, but currently setting a CW forces the status to be marked sensitive.

Reply:
> We can't easily differentiate between CWs that are there because it's food and the author is conscious of people who don't want to see food, and CWs that are there because it's adult content.

CWs are not just about “food” or “adult content”, but can cover anxiety-inducing topics or give a very concise summary of the topic of a long toot. Many legitimate and important toots have Content Warnings. Also, some instances may want “adult content” to trend.

> We err on the safe side here. Besides, browsing things you can't see and need to tap on is not the most satisfying experience.

Being exposed to anxiety-inducing or triggering content without a proper warning is not the most satisfying experience either, and I'd argue that not covering those topics at all could be considered as unsatisfying as well, but I guess that depends on your definition.

Reply:
Two other things to consider:

1. if toots with a CW cannot trend, then it will encourage some/most people not to use CWs because they want their toots to be popular
2. new users who browse trending toots will get used to not seeing CWs, so they won't use CWs themselves when tooting; which will lead overall to people not using CWs.